test/integration/scheduler/queue_test.go

```diff
@@ -170,6 +170,91 @@ func TestServiceAffinityEnqueue(t *testing.T) {
 	}
 }
 
+func TestCoreResourceEnqueue(t *testing.T) {
+	// Use zero backoff seconds to bypass backoffQ.
+	testCtx := testutils.InitTestSchedulerWithOptions(
+		t,
+		testutils.InitTestAPIServer(t, "core-res-enqueue", nil),
+		nil,
+		scheduler.WithPodInitialBackoffSeconds(0),
+		scheduler.WithPodMaxBackoffSeconds(0),
+	)
+	testutils.SyncInformerFactory(testCtx)
+	// It's intended to not start the scheduler's queue, and hence to
+	// not start any flushing logic. We will pop and schedule the Pods manually later.
+	defer testutils.CleanupTest(t, testCtx)
+
+	cs, ns, ctx := testCtx.ClientSet, testCtx.NS.Name, testCtx.Ctx
+	// Create one Node with a taint.
+	node := st.MakeNode().Name("fake-node").Capacity(map[v1.ResourceName]string{v1.ResourceCPU: "2"}).Obj()
+	node.Spec.Taints = []v1.Taint{{Key: "foo", Effect: v1.TaintEffectNoSchedule}}
+	if _, err := cs.CoreV1().Nodes().Create(ctx, node, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create Node %q: %v", node.Name, err)
+	}
+
+	// Create two Pods that are both unschedulable.
+	// - Pod1 is a best-efforts Pod, but doesn't have the required toleration.
+	// - Pod2 requests a large amount of CPU resource that the node cannot fit.
+	//   Note: Pod2 will fail the tainttoleration plugin b/c that's ordered prior to noderesources.
+	pod1 := st.MakePod().Namespace(ns).Name("pod1").Container("image").Obj()
+	pod2 := st.MakePod().Namespace(ns).Name("pod2").Req(map[v1.ResourceName]string{v1.ResourceCPU: "4"}).Obj()
```

Comment:
wrap the `pod2` with `Toleration` to make sure `pod2` will be failed with `noderesource` instead of the`taint`.

Reply:
It's intentional. Also explained in https://github.com/kubernetes/kubernetes/pull/105337#discussion_r718856954.

If pod2 goes with a mapped toleration, it'd fail due to NodeResource failure, and hence won't be triggered **at all**. However, in the case above, pod2 also failed due to TaintToleration, and hence looked like it should have been triggered. But why it's not? It's due to the `preCheckNode()` logic, which serves as the last gate to move a pod or not. So in this case, we used pod2 to cover the `preCheckNode()` logic.

BTW: I will come up with a pod3 case that failed VolumeBinding, but due to a tiny bug, I will raise a bug fix along with the test case.